### PR TITLE
Use dockers for linux builds

### PR DESCRIPTION
**Pull request by csujedihy**
## Description

Use onebranch linux build docker in public azure pipelines.

## Testing

CI

### CODE REVIEW

File: .azure/azure-pipelines.ci.yml [hunk 1]
```diff
@@ -356,15 +358,7 @@ stages:
       tls: openssl
       extraBuildArgs: -DisableLogs
       extraName: 'android'
-
```
Comment (csujedihy) [inline .azure/azure-pipelines.ci.yml:359]: This one is redundant. Already covered by github actions.

File: scripts/spin.ps1 [hunk 2]
```diff
@@ -127,7 +127,11 @@ if ($CodeCoverage) {
 $SpinQuic = $null
 if ($IsWindows) {
     $SpinQuic = Join-Path $RootArtifactDir "spinquic.exe"
-} elseif ($IsLinux -or $IsMacOS) {
+} elseif ($IsLinux) {
+    $SpinQuic = Join-Path $RootArtifactDir "spinquic"
+    $SpinQuicRootDir = Split-Path $SpinQuic -Parent
+    $env:LD_LIBRARY_PATH = "${SpinQuicRootDir}:$env:LD_LIBRARY_PATH"
```
Comment (nibanks) [inline scripts/spin.ps1:133]: Why is this needed only for docker builds?

Comment (csujedihy) [reply scripts/spin.ps1:133]: This is still a mystery to me. When I first ran into `libmsquic.so.2: cannot open shared object file` in quicinterop and spinquic, the pipeline was run against the commit, [e079de8](https://github.com/microsoft/msquic/pull/3358/commits/e079de803036340357909911bb103978e09c9a7a), that doesn't even touch the build jobs that quicinterop and spinquic depend on. In other words, making irrelevant build tasks use container somewhat changed the behavior of quicinterop and spinquic, which I don't quite understand.

Comment (csujedihy) [reply scripts/spin.ps1:133]: And btw, we don't set rpath for quicinterop and spinquic. So, they are not supposed to find shared libs from the origin folder. I don't know why they even worked before my PR.

Comment (nibanks) [reply scripts/spin.ps1:133]: Maybe because everything is in the same place exactly for build and test machines before?

Comment (csujedihy) [reply scripts/spin.ps1:133]: I honest don't know.
